libraries/AP_AngleSensor/AP_AngleSensor.h

```diff
@@ -0,0 +1,111 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "AP_AngleSensor_config.h"
+
+#if AP_ANGLESENSOR_ENABLED
+
+#include <AP_Common/AP_Common.h>
+#include <AP_Param/AP_Param.h>
+#include <AP_Math/AP_Math.h>
+#include "AP_AngleSensor_Params.h"
+
+#define ANGLE_SENSOR_TYPE_NONE 0
+
+class AP_AngleSensor_Backend; 
+class AP_AngleSensor_AS5048B;
+ 
+class AP_AngleSensor
+{
+public:
+    friend class AP_AngleSensor_Backend;
+    friend class AP_AngleSensor_AS5048B;
+
+    AP_AngleSensor(void);
+
+    /* Do not allow copies */
+    CLASS_NO_COPY(AP_AngleSensor);
+
+    // get singleton instance
+    static AP_AngleSensor *get_singleton() {
+        return _singleton;
+    }
+
+    // AngleSensor driver types
+    enum AngleSensor_Type : uint8_t {
+        ANGLESENSOR_TYPE_NONE          =   ANGLE_SENSOR_TYPE_NONE,
+        ANGLESENSOR_TYPE_AS5048B       =   1,
+    };
```

Comment:
```suggestion
    enum class TYPE : uint8_t {
        NONE          =   ANGLE_SENSOR_TYPE_NONE,
        AS5048B       =   1,
    };
```
Enum class's are preferred, you would then use like this: `TYPE::AS5048B`. Externally (its public so I guess there is a user somewhere) it would be `AP_AngleSensor::TYPE::AS5048B` so you still have the same descriptiveness of `ANGLESENSOR_TYPE_AS5048B`